crates/matrix-sdk/src/event_cache/linked_chunk/updates.rs

```diff
@@ -61,23 +61,31 @@ pub enum Update<Item, Gap> {
     /// A chunk has been removed.
     RemoveChunk(ChunkIdentifier),
 
-    /// Items are inserted inside a chunk of kind Items.
-    InsertItems {
-        /// [`Position`] of the items.
-        at: Position,
+    /// Items are pushed inside a chunk of kind Items.
+    PushItems {
+        /// The [`Position`] of the items.
+        ///
+        /// This value is given to prevent the need for position computations by
+        /// the update readers. Items are pushed, so the positions should be
+        /// incrementally computed from the previous items, which requires the
+        /// reading of the last previous item. With `position_hint`, the update
+        /// readers no longer need to do so.
+        position_hint: Position,
```

Comment:
I've had a hard time finding the best naming here. `at` could imply it can be _at any position_ within the chunk, so possibility before or in the middle of existing items, while it's impossible and will never happen. It's a _push_. The `Position` is not required. It's here only to save computations from the consumers. For example, if we need to insert in a database, we need to first read the items, sort them, and find the last items; from here we can compute the position for the newly pushed items. This reading + sorting can be costly, hence `position_hint`, it's a pre-computed position.

Naming it `at` sounds bad to me. Naming it `position` creates a feeling of inconsistency with the rest of the code. That's why I've decided to use `position_hint`. It may not be the best idea I had, but in my mental scheme, it works 😛.

Reply:
I don't get it; if the `position_hint` is always used, and we always insert at the `position_hint` position, how is that conceptually different from the position `at` which we want to insert?

Reply:
Renamed to `at`.